spec/lib/miq_expression/tag_spec.rb

```diff
@@ -2,19 +2,54 @@
 
 RSpec.describe MiqExpression::Tag do
   describe ".parse" do
-    it "with managed-field" do
-      field = "managed.location"
-      expect(described_class.parse(field)).to eq(described_class.new(nil, "/location"))
-    end
-
     it "with model.managed-in_field" do
       field = "Vm.managed-service_level"
-      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/service_level"))
+      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/service_level", []))
     end
 
     it "with model.last.managed-in_field" do
       field = "Vm.host.managed-environment"
-      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/host"))
+      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/environment", ['host']))
+    end
+
+    it "with model.managed-in_field" do
+      field = "Vm.managed-service_level"
+      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/service_level", []))
+    end
+
+    it "with model.associations.associations.managed-in_field" do
+      field = "Vm.service.user.managed-service_level"
+      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/service_level", %w(service user)))
+    end
+
+    it "with model.associations.managed-in_field" do
+      field = "service.user.managed-service_level"
```

Comment:
Good point. Yes in `MiqExpression` we have it only with upper case. I realized that I needed this case only for clean up value2tag (maybe removing). When input to this method is tag 
(for example 'Service.user.managed-service_level') then `preprocess_managed_tag`converts 
 it to `user.managed-service_level` and then it goes thru value2tag and I wanted to use MiqExpression::Tag here. 

So yes temporarily I need it here now but in follow-up PRs I will tend to remove 
- `value2tag`
- `preprocess_managed_tag`
- remove this case from `MiqExpression::Tag`( and change REGEX ...)